help: Improve relative settings links for documentation on bots.

drrosa opened the pull request:
I noticed that some of the documentation on bots is not accurate, e.g. https://zulip.com/help/edit-a-bot:  
<img width="274" alt="image" src="https://user-images.githubusercontent.com/2343554/213845138-4f8f0be2-7021-47e4-ab2f-6709b4f5f3da.png">
<img width="302" alt="image" src="https://user-images.githubusercontent.com/2343554/213845144-ce590187-8776-4295-85ba-bde2a13f1b00.png">

due to the relative links to the **Bots** tab, e.g. https://chat.zulip.org/help/edit-a-bot:
<img width="232" alt="image" src="https://user-images.githubusercontent.com/2343554/214078671-dc9e4062-4cea-4ba6-b60d-f78c6dcdc4f1.png">
<img width="234" alt="image" src="https://user-images.githubusercontent.com/2343554/214078787-f04a444b-10ae-474f-8e81-75f10dd2fbee.png">

This PR fixes the documentation generated from the Markdown macros `{settings_tab|your-bots}` and `{settings_tab|bot-list-admin}` to match the text labels in the Zulip UI and improves the text of relative links to explicitly say if we are referring to the **Bots** tab of the **Personal** or **Organization** settings menu.

This PR does some refactoring in help_settings_links.py, making it easier to document special cases of Zulip menu settings, like tabs with the same name in both Personal and Organization settings.

 Follow-up to #23256.

**Screenshots and screen captures:**

<img width="493" alt="image" src="https://user-images.githubusercontent.com/2343554/213845001-e484ec5c-a8f7-4623-a986-5588b12481d1.png">

<img width="472" alt="image" src="https://user-images.githubusercontent.com/2343554/213845062-3aa39653-70ab-4b66-8397-544ded2694d0.png">

<details>
<summary>Self-review checklist</summary>

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans.
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Links.
</details>

alya commented:
@timabbott Do you want to just review this one directly? I haven't tested.

timabbott commented on zerver/lib/markdown/help_settings_links.py:
```diff
@@ -118,6 +118,11 @@ def getMarkdown(setting_type_name: str, setting_name: str, setting_link: str) ->
     )
+    if setting_name == "Bots":
+        if setting_type_name == "Manage organization":
+            setting_type_name = "Organization settings"
```
I'm not sure this makes sense -- the gear menu item is called "Manage organization" if you're an administrator, and "Organization settings" otherwise -- I'm not sure it makes sense to have this conditional naming for it? Like, if we called it "Manage organization" unconditionally, I think that seems fine -- administrators don't see the string "organization settings" in the app, and this is a feature documented as for administrators only...

drrosa replied:
The thing is that without this conditional the instruction would say:
> Navigate to the **Bots** tab of the **Manage Organization** menu.

The reasoning for using "Organization settings" is to refer to the following part of the app:

<img width="1027" alt="image" src="https://user-images.githubusercontent.com/2343554/214171369-af581764-cf5f-4329-aa26-bbefea108462.png">

which does have the strings "Organization" and "Settings", and both administrators and users see the same strings.

alya replied:
Yeah, I don't want to call the *menu* itself "Manage organization".

We should potentially add a word on an icon before "Settings" to make it more clear what you are looking at, in case you don't notice the tabs.

alya replied:
I suppose we could also drop the special naming in the gear menu, and just call it "Organization settings".

timabbott replied:
Yeah, just calling it "Organization settings" everywhere might be the simplest solution here. Worth discussing in #design; I think the "Manage organization" name was a slightly nice touch before we had any documentation, but it's super minor and creates this kind of awkwardness, especially now that we have roles like "moderator" that are kinda in between.

alya replied:
https://chat.zulip.org/#narrow/stream/101-design/topic/.22Manage.20organization.22.20menu.20item/near/1496114

timabbott commented:
I guess this should be closed or adjusted now that we've standardized on the "Organization settings" name.

drrosa commented:
> I guess this should be closed or adjusted now that we've standardized on the "Organization settings" name.

I removed the awkward conditional since it's no longer needed.

Also, as @alya noted here https://github.com/zulip/zulip/pull/23256#issuecomment-1281591308 I think this PR is still useful to fix the following and to explicitly say which Bots tab we're talking about:
<img width="274" alt="image" src="https://user-images.githubusercontent.com/2343554/213845138-4f8f0be2-7021-47e4-ab2f-6709b4f5f3da.png">
<img width="302" alt="image" src="https://user-images.githubusercontent.com/2343554/213845144-ce590187-8776-4295-85ba-bde2a13f1b00.png">

drrosa commented:
> I think it would be kinda nice if we could avoid the "your" in "Go to your Bots" being part of the link; if we want to do that, we'll probably need to do a separate project to refactor `help_settings_links.py` to offer more flexibility than `return f"1. Go to [{setting_name}]({setting_link})."`. What I'd consider doing is changing the `link_mapping` structure from a map to tuples:
> 
>  `profile": ["Personal settings", "Profile", "/#settings/profile"]`
> 
> to a Python dataclass or some other structure that makes it reasonably clean to add additional fields. Probably a good thing to discuss in #documentation to get others' ideas before committing to a technical plan.

_Originally posted by @timabbott in https://github.com/zulip/zulip/issues/23256#issuecomment-1282749917_

btw, do we still want to put some more effort into  a separate project to refactor `help_settings_links.py`?

alya commented:
@timabbott Re-adding the integration review label; see @drrosa 's comment above.

drrosa commented:
> Re-adding the integration review label

It seems like there was an issue adding the label.

@zulipbot add "integration review"

timabbott commented on zerver/lib/markdown/help_settings_links.py:
```diff
@@ -111,17 +111,18 @@
-        setting_reference=f"**{setting_name}**",
+        setting_reference=setting_name,
     )
+    return relative_link_markdown if relative_settings_links else setting_markdown
```
Code style wise, I prefer multiple return statements if normal if/else clauses; it's easier for one's eyes to see the control code of the function. 

In this particular case the performance issue doesn't matter, but you're also computing both answers with this implementation, and that's something to avoid from a coding habits standpoint.

timabbott commented on zerver/lib/markdown/help_settings_links.py:
```diff
@@ -122,6 +122,9 @@ def getMarkdown(setting_type_name: str, setting_name: str, setting_link: str) ->
         setting_type_name=setting_type_name,
         setting_reference=setting_name,
     )
+    if setting_name == "Bots":
```
Let's add a comment in this block explaining that we need special text here because Bots is a label that appears in both tabs.

drrosa commented:
@timabbott Thanks for the feedback! Makes sense. :)
I dropped the second commit, added a comment, and fixed the code style in response to your code review. Also updated the PR description.

timabbott commented:
Merged, thanks @drrosa!
